### PR TITLE
layout dataStore api implementation logic

### PR DESCRIPTION
**Pull request by NicoPotamus**
Can you just verify that my logic is coherent. If so I'll implement what I can in the class and controller.

### CODE REVIEW

File: src/engine/dataapi/DataStore.java [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+	
     /**
-     * Receives a data request and processes it.
+     * Receives a data request for output and processes it.
+     * we're gonna pass an id as well that we'll make in the controller to have a reference to our obj in the db
      */
     @Override
-    public DataStream receiveDataRequest(DataStream data) {
-        return null;
+    public DataStream receiveDataRequest(DataStream newData, String id) {
```
Comment (dgunther2001) [inline src/engine/dataapi/DataStore.java]: This needs to do its job and actually store the value

File: src/engine/dataapi/DataStream.java [hunk 2]
```diff
@@ -1,8 +1,32 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
 /**
  * Data stream passed to and from the controller and data storage.
  */
 public class DataStream implements ProtoDataStream{
 
+    private List<String> data = new ArrayList<>();
```
Comment (dgunther2001) [inline src/engine/dataapi/DataStream.java:13]: Data stream should be an individual value, not an away. We should wrap the final request back to the controller after all data has been processed into a completely separate interface

Comment (dgunther2001) [reply src/engine/dataapi/DataStream.java:13]: Actually nvm, we discussed this and decided on single element lists

File: src/engine/dataapi/DataStream.java [hunk 3]
```diff
@@ -1,8 +1,32 @@
...
+    /**
+     * Pass in a list of data to be stored.
+     * @param data
+     */
+    public DataStream(List<String> data) {
+        this.data = data;
+
+    }
+
+    public List<String> getData() {
```
Comment (dgunther2001) [inline src/engine/dataapi/DataStream.java:40]: Not: add javadoc

File: src/engine/dataapi/DataStream.java [hunk 4]
```diff
@@ -1,8 +1,32 @@
...
+        return this.data;
+    }
+    
+    public void setData(List<String> data) {
```
Comment (dgunther2001) [inline src/engine/dataapi/DataStream.java:55]: Nit: add javadoc

File: src/engine/dataapi/DataStore.java [hunk 5]
```diff
@@ -1,16 +1,27 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
 /**
  * Receiving end of the data storage API.
  */
 public class DataStore implements ProtoDataStore{
-
+	List<Integer> data;
```
Comment (dgunther2001) [inline src/engine/dataapi/DataStore.java]: Nit: add javadoc

File: src/engine/dataapi/ProtoDataStream.java [hunk 6]
```diff
@@ -5,4 +5,5 @@
  */
 public interface ProtoDataStream {
     
+
```
Comment (dgunther2001) [inline src/engine/dataapi/ProtoDataStream.java:8]: Nit: add javadoc

**Review by dgunther2001 — CHANGES_REQUESTED** [review]
The data processing needs to actually occur. The data storage request is returning null. In addition, this needs to be connected to the controller. If you make the adjusted changes, I can connect it to the controller over the weekend.

File: src/engine/dataapi/DataStore.java (hunk 1 above)
Comment (Nikolas023) [reply src/engine/dataapi/DataStore.java]: I agree!

File: src/engine/dataapi/DataStore.java [hunk 7]
```diff
@@ -1,17 +1,75 @@
...
+    			 fileObject = newData;
+    			 objOut.writeObject(fileObject);
+    			 return fileObject;
+    		 }
+    		 else {//if a list was started add to it
+    			 fileObject.append(newData);
+    			 objOut.writeObject(fileObject);
+    			 return fileObject;
+    		 }
+    	}
```
Comment (dgunther2001) [inline src/engine/dataapi/DataStore.java:64]: Bring catch up next to '}'

File: src/engine/dataapi/DataStore.java [hunk 8]
```diff
@@ -1,17 +1,75 @@
...
+    				 fileObject = (DataStream) objIn.readObject();
+    				 if(fileObject.getId() == id) {
+    					  break;
+    				 }
+    			 }
+    			 catch (IOException | ClassNotFoundException e) {
+    				 break;
+    			 }
+    		 }
+    		 if(fileObject == null) {//if list wasn't started make a new one
```
Comment (dgunther2001) [inline src/engine/dataapi/DataStore.java:54]: Bring if up to '}'

File: src/engine/dataapi/DataStore.java [hunk 9]
```diff
@@ -1,17 +1,75 @@
...
+        	ObjectOutputStream objOut = new ObjectOutputStream(fileOut);
+        	
+    		 while(true) {//check file for existing entry
+    			 try {
+    				 fileObject = (DataStream) objIn.readObject();
+    				 if(fileObject.getId() == id) {
+    					  break;
+    				 }
+    			 }
+    			 catch (IOException | ClassNotFoundException e) {
```
Comment (dgunther2001) [inline src/engine/dataapi/DataStore.java:50]: Bring catch up to '}'

File: src/engine/dataapi/DataStore.java [hunk 10]
```diff
@@ -1,17 +1,75 @@
...
+     * we're gonna pass an id as well that we'll make in the controller to have a reference to our obj in the db
+     * @param newData is the data to be appended and/or stored
+     * @param id is the id of the object/list
+     * @param file is the user-specified file
      */
     @Override
-    public DataStream receiveDataRequest(DataStream data) {
-        return null;
+    public DataStream receiveDataRequest(DataStream newData, String id, File file) {
+        // search db or file for id
```
Comment (dgunther2001) [inline src/engine/dataapi/DataStore.java:27]: We're not rrying serialize things yet, and this adds some over complication as we're not using a raw string anymore which kind of breaks the system.

**Review by dgunther2001 — APPROVED** [review]
This needs to be connected to the controller, and tests need to be reenabled, but I will do that today and make a pr. They are all quick fixes. The data store method also needs an overloaded version that deals with the simple string representation because right now the file objects break the system. Nonetheless, it doesn't introduce any breaking changes, so I'm gonna pull it in and make a few changes in a new branch.

**Pull request merged**